### ISSUE TITLE
Line-level suppression of reportUnnecessaryTypeIgnoreComment

### ISSUE THREAD

**Issue by darlingm** [microsoft/pyright#9084]
**Describe the bug**

With the code below, mypy with `disallow_any_expr = True` gives:

`error: Expression type contains "Any" (has type overloaded function)  [misc]`

I can disable this mypy error using `# type: ignore`.  But, then pyright says:

`warning: Unnecessary "# type: ignore" comment (reportUnnecessaryTypeIgnoreComment)`

I realize that although I'm using pylance through VSCode, pyright doesn't have access to the problems window to know it's necessary for a different checker.  So, pyright's warning is legitimate, but now I need to disable that.  I've tried the following, all of which cause pyright to increase the line to 2 warnings, saying both are unnecessary!  😂

* `# type: ignore  # pyright: ignore`
* `# pyright: ignore  # type: ignore`
* `# type: ignore  # type: ignore`
* `# type: ignore  # pyright: ignore [reportUnnecessaryTypeIgnoreComment]`

I can put this at the top of the file, but I'd rather suppress it on the single line:

`# pyright: reportUnnecessaryTypeIgnoreComment=false`

**Code**

With SQLAlchemy 2.0.35:

```python
from sqlalchemy.orm import declarative_mixin, Mapped, mapped_column

@declarative_mixin  # type: ignore
class _TimestampedMixin:
    created_at: Mapped[datetime] = mapped_column(nullable=False)
    updated_at: Mapped[datetime] = mapped_column(nullable=False)
```

(Yes, in this situation the mapped_columns are unnecessary since `nullable=False` is inferred, but then that creates a reportUninitializedInstanceVariable warning, and I decided to initialize these in this sample code to keep things focused.)

**VS Code extension or command-line**
I am running pyright integrated into Pylance.

**Comment by erictraut** [comment]
No, you can't suppress an "unnecessary type ignore comment" error by adding another type ignore comment. :)

If you're using mypy as a type checker, then using `reportUnnecessaryTypeIgnoreComment` in pyright or pylance is probably not very practical. Unfortunately, mypy doesn't have a mypy-specific way to suppress its errors like pyright does. I've lobbied for mypy maintainers to add a `# mypy: ignore` comment, but they haven't been receptive to the idea so far.
